include/zephyr/bluetooth/bluetooth.h

```diff
@@ -514,6 +514,13 @@ enum {
 	/**
 	 * @brief Advertise as connectable.
 	 *
+	 * @deprecated Use @ref BT_LE_ADV_OPT_CONN instead.
```

Comment:
This will only mark it as deprecated in documentation, right? It won't generate any build-time warnings. 

Can we apply `__deprecated` to the deprecated enum values similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/data/json.h#L36-L39 ?